kafka-plugins-0.10/src/main/java/io/cdap/plugin/source/KafkaConfig.java

```diff
@@ -351,7 +374,7 @@ public Schema getMessageSchema(FailureCollector collector) {
 
     // set default initial partitions
     for (Integer partition : partitionsToRead) {
-      partitionOffsets.put(new TopicPartition(topic, partition), defaultInitialOffset);
+      partitionOffsets.put(new TopicPartition(topic, partition), getDefaultInitialOffset());
```

Comment:
We can call method _getDefaultInitialOffset()_ outside of loop declaring it as final.